build: Enable support of musl target

eryugey opened the pull request:
Fix errors when compiling with musl target, as musl libc has different
variable type and padding.

Fixes: #17 
Signed-off-by: Liu Jiang <gerry@linux.alibaba.com>

jiangliu committed:
build: Enable support of musl target

Fix errors when compiling with musl target, as musl libc has different
variable type and padding.

Signed-off-by: Liu Jiang <gerry@linux.alibaba.com>
Signed-off-by: Eryu Guan <eguan@linux.alibaba.com>

eryugey commented:
@sameo @sboeuf Could you please take a look? Thanks!

sameo commented:
@eryugey The changes look good, but could you please also modify the github action CI so that we no longer continue on error for the musl build and unit tests? Just remove those lines:
https://github.com/cloud-hypervisor/vhost/blob/dragonball/.github/workflows/main.yml#L36
https://github.com/cloud-hypervisor/vhost/blob/dragonball/.github/workflows/main.yml#L43

eryugey commented:
> @eryugey The changes look good, but could you please also modify the github action CI so that we no longer continue on error for the musl build and unit tests? Just remove those lines:
> https://github.com/cloud-hypervisor/vhost/blob/dragonball/.github/workflows/main.yml#L36
> https://github.com/cloud-hypervisor/vhost/blob/dragonball/.github/workflows/main.yml#L43

Sure and done, and the tests have passed now.

sameo commented:
@eryugey Thanks. Sorry to be a pain in the butt, but could you please move the github action changes into their own commit? This will make our lives easier when upstreaming that to rust-vmm.

eryugey committed:
github: test musl builds

Now we enabled musl build, let's update github action CI accordingly so
that we no longer continue on error for the musl build and unit tests.

Signed-off-by: Eryu Guan <eguan@linux.alibaba.com>

eryugey commented:
Thanks @sameo !
